table/block_based/filter_block.h

```diff
@@ -137,7 +137,8 @@ class FilterBlockReader {
       const Slice ukey = iter->ukey;
       const Slice ikey = iter->ikey;
       GetContext* const get_context = iter->get_context;
-      if (!KeyMayMatch(prefix_extractor->Transform(ukey), prefix_extractor,
+      if (prefix_extractor->InDomain(ukey) &&
+          !KeyMayMatch(prefix_extractor->Transform(ukey), prefix_extractor,
```

Comment:
Yes, the call to ```KeyMayMatch``` should be ```PrefixMayMatch``` instead. I'll change it. As a practical matter, it may not make much difference. Also, the ```InDomain``` check is still required as ```PrefixMayMatch``` implementation does not check it.